AutomatedTesting/Gem/PythonTests/scripting/scripting_utils/scripting_tools.py

```diff
@@ -0,0 +1,66 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import os
+from editor_python_test_tools.utils import TestHelper as helper
+from editor_python_test_tools.utils import Report
+from PySide2 import QtWidgets, QtTest, QtCore
+import editor_python_test_tools.pyside_utils as pyside_utils
+import azlmbr.legacy.general as general
+import azlmbr.editor as editor
+import azlmbr.bus as bus
+from scripting_utils.scripting_constants import (SCRIPT_CANVAS_UI, ASSET_EDITOR_UI, SCRIPT_EVENT_UI, NODE_PALETTE_UI,
+                                                 NODE_PALETTE_QT, TREE_VIEW_QT, EVENTS_QT, DEFAULT_METHOD_NAME,
+                                                 SEARCH_FRAME_QT, SEARCH_FILTER_QT, EVENT_NAME_QT, NAME_STRING,
+                                                 PARAMETERS_QT, SAVE_STRING, SAVE_ASSET_AS, WAIT_TIME_3)
+
+
+def save_file(self, file_path):
```

Comment:
Since we're using libraries that aren't exactly clear in what we're doing without context, can we please add some definition docstring block comments to each of these definitions so if somebody goes to use the tool they can easily see what it's doing.

Additionally, since we're moving these into a utils file, we should change the name of `save_file()` to something less generic and more meaningful such as `save_sc_graph()` or whatever the actual purpose is.

EG:

```
def save_sc_graph(self, file_path):
"""
Save a script canvas graph by doing things then other things

:param file_path: The path to save the graph
:return bool: Returns whether or not the the graph saved correctly
"""

```

Reply:
renamed function and added comments to functions to make their intents clearer to users in the future. 

Also had a brief conversation about making a QT model that all these scripts would use to guarantee qt objects exist and these functions are more stable/easier to use.